lib/cubit/payments/payments_state.dart

```diff
@@ -0,0 +1,69 @@
+import 'dart:convert';
+
+import 'package:flutter_breez_liquid/flutter_breez_liquid.dart';
+import 'package:l_breez/cubit/payments/models/models.dart';
+
+class PaymentsState {
+  final List<PaymentData> payments;
+  final PaymentFilters paymentFilters;
+
+  const PaymentsState({required this.payments, required this.paymentFilters});
+
+  PaymentsState.initial() : this(payments: [], paymentFilters: PaymentFilters.initial());
+
+  PaymentsState copyWith({
+    List<PaymentData>? payments,
+    PaymentFilters? paymentFilters,
+  }) {
+    return PaymentsState(
+      payments: payments ?? this.payments,
+      paymentFilters: paymentFilters ?? this.paymentFilters,
+    );
+  }
+
+  List<PaymentData> get filteredPayments {
+    List<PaymentData> filteredPayments = payments;
+    // Apply date filters, if there's any
+    if (paymentFilters.fromTimestamp != null || paymentFilters.toTimestamp != null) {
+      filteredPayments = payments.where((paymentData) {
+        final fromTimestamp = paymentFilters.fromTimestamp;
+        final toTimestamp = paymentFilters.toTimestamp;
+        final milliseconds = paymentData.paymentTime.millisecondsSinceEpoch;
+        if (fromTimestamp != null && toTimestamp != null) {
```

Comment:
This only filters when both `to` and `from` timestamps are set, doesn't filter if only one is set.

Is this the intended behavior?

Reply:
> This only filters when both `to` and `from` timestamps are set, doesn't filter if only one is set.
> 
> Is this the intended behavior?

Yes, date picker for filtering payments requires a `from` & `to`. They are `paymentTime` of the first item on the payments list, and `DateTime.now()` by default respectively.